### PR TITLE
Status Menunode synchronization with peerings information

### PR DESCRIPTION
**Pull request by AbakusW**
# Description

This PR operates a refactoring on the *Status* MenuNode (which shows some metrics on the tray menu) and introduces the possibility for the Agent to manage the status of active peerings (read only).

* STATUS MenuNode
  * code refactoring for a cleaner and more efficient workflow
  * added an explicit *peers* counter to aggregate all data inside the Status
  * the tray agent *label* (small text beside the tray icon) is now used to dynamically display the number of active peerings (with a distinction between **incoming** and **outgoing** ones)

* Peerings management (read only)
  * The AgentController now handles also events on the *ForeignCluster* CRD regarding the establishment and disconnection of peerings
  * The Indicator listens to these two kinds of events and reacts by:
    * notifying the event (acording to current notification settings)
    * updating STATUS data
    * signalling the event inside the "*peers* list" for the corresponding peer; currently a ✔️ is used to indicate whether a peering in a given direction is active
    * the function handlers regarding the Advertisements have been removed, since the updates about peerings are now acquired from the ForeignCluster

### NOTE: 

The peerings feature implemented in this PR is an early version. In a very next PR there will be a code refactoring for the **NotifyChannel**s and Indicator **Listener**s, thus enabling a more efficient and compact handling of these events.

# How Has This Been Tested?

Since this is not a definitive version and a relevant portion of code will be soon updated, the tests only cover the STATUS refactoring. New tests will be added in the updating PR. Some live tests have been conducted locally.

### CODE REVIEW

**Review by mlavacca — APPROVED** [review]

File: internal/tray-agent/agent/client/advertisementController.go [hunk 1]
```diff
@@ -8,11 +8,7 @@ import (
 
 //createAdvertisementController creates a new CRDController for the Liqo Advertisement CRD.
 func createAdvertisementController(kubeconfig string) (*CRDController, error) {
-	controller := &CRDController{
-		addFunc:    advertisementAddFunc,
-		updateFunc: advertisementUpdateFunc,
-		deleteFunc: advertisementDeleteFunc,
-	}
```
Comment (aleoli) [inline internal/tray-agent/agent/client/advertisementController.go:15]: why did you remove this advertisement function?

Comment (AbakusW) [reply internal/tray-agent/agent/client/advertisementController.go:15]: In this new version, the main point of access is the foreign cluster. It's the ForeignClusterController's cache that monitors with its handlers the changes on a FC, like the 'joined' field for the peerings.

I considered at least for this moment to remove the handling of advertisements events, since a notification would now be useless and there is no associated behavior to these changes.

Moreover, in the next PR I'll bring some refactoring to the Listener mechanism, and so there would be additional useless code to refactor

**Review by aleoli — APPROVED** [review]
